HW3/HW3.py

```diff
@@ -0,0 +1,65 @@
+import sys
+
+from pyspark import StorageLevel
+from pyspark.sql import SparkSession
+
+
+def main():
+    spark = SparkSession.builder.master("local[*]").getOrCreate()
+
+    user = input("Enter your mariadb user: ")
+    password = input("Enter the associated password: ")
+
+    batter_counts_df = (
+        spark.read.format("jdbc")
+        .option("url", "jdbc:mysql://localhost:3306/baseball")
+        .option("driver", "org.mariadb.jdbc.Driver")
+        .option("dbtable", "batter_counts")
+        .option("user", user)
+        .option("password", password)
+        .load()
+    )
+
+    batter_counts_df.createOrReplaceTempView("batter_counts")
+    batter_counts_df.persist(StorageLevel.DISK_ONLY)
+
+    game_df = (
+        spark.read.format("jdbc")
+        .option("url", "jdbc:mysql://localhost:3306/baseball")
+        .option("driver", "org.mariadb.jdbc.Driver")
+        .option("dbtable", "game")
+        .option("user", user)
+        .option("password", password)
+        .load()
+    )
+
+    game_df.createOrReplaceTempView("game")
+    game_df.persist(StorageLevel.DISK_ONLY)
+
+    query = """
+    SELECT a.game_id,
+           a.batter,
+           CASE WHEN sum(c.atBat) > 0 THEN
+           round(sum(c.Hit) / sum(c.atBat), 3)
```

Comment:
I think we are supposed to use a transform for this calculation but I could be wrong